### PR TITLE
[ZEPPELIN-632][DOC] Document steps to contribute a new interpreter

### PR DESCRIPTION
**Pull request by felixcheung**
### What is this PR for?
Instead of going through a list of items one by one in PR review, we should have a checklist of TODOs for a potential interpreter contributor to review and follow 

### What type of PR is it?
Documentation

### Todos
* [x] - Update doc

### Is there a relevant Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-632

### How should this be tested?
N/A

### Screenshots (if appropriate)
N/A

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? This is.

### CODE REVIEW

**Comment by felixcheung** [comment]
any comment?

File: docs/development/writingzeppelininterpreter.md [hunk 1]
```diff
@@ -138,15 +150,24 @@ You can only omit your interpreter group when your interpreter group is selected
...
+
+ - First, check out the general contribution guide [here](./howtocontributewebsite.html).
+ - Follow the steps in "Make your own Interpreter" section above.
+ - Add your interpreter as in the "Configure your interpreter" section above; also add it to the example template [zeppelin-site.xml.template](https://github.com/apache/incubator-zeppelin/blob/master/conf/zeppelin-site.xml.template).
+ - Add tests! They are run by Travis for all changes and it is important that they are self-contained.
+ - Include your interpreter as a module in [`pom.xml`](https://github.com/apache/incubator-zeppelin/blob/master/pom.xml).
+ - Add documentation on how to use your interpreter under `docs/interpreter/`. Follow the Markdown style as this  [example](https://github.com/apache/incubator-zeppelin/blob/master/docs/interpreter/elasticsearch.md). Make sure you list config settings and provide working examples on using your interpreter in code boxes in Markdown. Link to images as appropriate (images should go to `docs/assets/themes/zeppelin/img/docs-img/`). And add a link to your documentation in the navigation menu (`docs/_includes/themes/zeppelin/_navigation.html`).
```
Comment (AhyoungRyu) [inline docs/development/writingzeppelininterpreter.md]: There is a extra space between "~ style as this" and "[example]".

File: docs/development/writingzeppelininterpreter.md [hunk 2]
```diff
@@ -138,15 +150,24 @@ You can only omit your interpreter group when your interpreter group is selected
...
+ - Most importantly, ensure licenses of the transitive closure of all dependencies are list in [license file](https://github.com/apache/incubator-zeppelin/blob/master/zeppelin-distribution/src/bin_license/LICENSE).
+ - Commit your changes and open a Pull Request on https://github.com/apache/incubator-zeppelin; check to make sure Travis CI build is passing.
```
Comment (AhyoungRyu) [inline docs/development/writingzeppelininterpreter.md]: How about adding a link for this [https://github.com/apache/incubator-zeppelin](https://github.com/apache/incubator-zeppelin) ?

File: docs/development/writingzeppelininterpreter.md (hunk 1 above)
Comment (felixcheung) [inline docs/development/writingzeppelininterpreter.md]: There is something strange about this - I checked it's not showing up with double space
image

File: docs/development/writingzeppelininterpreter.md (hunk 2 above)
Comment (felixcheung) [inline docs/development/writingzeppelininterpreter.md]: done

**Comment by AhyoungRyu** [comment]
@felixcheung Looks good!

**Comment by felixcheung** [comment]
merging if no more comment

**Pull request closed without merging**